src/entities/DFGVariable.jl

```diff
@@ -16,58 +16,92 @@ Data container for solver-specific data.
 Fields:
 $(TYPEDFIELDS)
 """
-mutable struct VariableNodeData{T<:InferenceVariable}
-    val::Array{Float64,2}
-    bw::Array{Float64,2}
-    BayesNetOutVertIDs::Array{Symbol,1}
-    dimIDs::Array{Int,1} # Likely deprecate
+mutable struct VariableNodeData{T<:InferenceVariable, P}
+    val::Vector{P}
+    bw::Vector{Vector{Float64}}
```

Comment:
Why this structure? I also use it as a Matrix for the covariance in parametric.

Reply:
oh i just duplicated val to bw.  We can leave bw as Matrix for now.  Sorry I'll change that back